### PR TITLE
feat: menu/other UI from header avatar with username

### LINKED ISSUE #105: Feature: Feature: menu/other UI from header avatar with username
### Type of feature

🍕 Feature

### Current behavior

When you login to the app the GitHub profile avatar appears in the top right-hand corner. The avatar doubles as a logout button and that are not clear to the user who may be trying to click through a profile..

### Suggested solution

It would be ideal to make the profile a dropdown so the user can get details of the app. This was done in [opensauzed.pizza](https://opensauced.pizza/) by @blackgirlbytes https://github.com/open-sauced/open-sauced/pull/1129/files

Screen Shot 2022-01-23 at 7 53 31 PM

### Additional context

Please use Taildwind to accomplish this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nulfrost**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR adds a dropdown menu when the user clicks on their avatar on the website.

## Related Tickets & Documents

Fixes #105 

## Mobile & Desktop Screenshots/Recordings
Screen Shot 2022-01-24 at 8 24 04 PM

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

Possibly double check to make sure the dropdown is working, there's still that issue from #76 where the app breaks after login which seems to be only happening locally. 

## [optional] What gif best describes this PR or how it makes you feel?

coolcoolcool

### CODE REVIEW

**Comment by 0-vortex** [comment]
> # PR Compliance Checks
> Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> 
> ## Watched Files
> This pull request modifies specific files that require careful review by the maintainers.
> 
> Files Matched
> 
> * npm-shrinkwrap.json
> * package.json

You reverted the `lockfileVersion` to 2 in this PR, please use `npm>8` and restore the shrinkwrap and `package.json` from main, then reinstall the menu :D

File: public/favicon.ico [hunk 1]
```diff
@@ -0,0 +1 @@
+https://rawcdn.githack.com/snowpackjs/astro/main/examples/starter/public/favicon.ico
```
Comment (0-vortex) [inline public/favicon.ico:1]: we could use the open-sauced one here, the link provided is a 404

**Comment by nulfrost** [comment]
> > # PR Compliance Checks
> > Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> > ## Watched Files
> > This pull request modifies specific files that require careful review by the maintainers.
> > Files Matched
> > 
> > * npm-shrinkwrap.json
> > * package.json
> 
> You reverted the `lockfileVersion` to 2 in this PR, please use `npm>8` and restore the shrinkwrap and `package.json` from main, then reinstall the menu :D

Ah, never meant to revert, what was the command to restore again?

> we could use the open-sauced one here, the link provided is a 404

cool cool, I'll grab it from the main repo

> this is not the way we should do it with astro/vite. please remove these 2 lines in favour of `astro.config.mjs`:
> 
> ```js
> {
>   vite: {
>     define: {
>       '__APP_VERSION__': JSON.stringify(process.env.npm_package_version),
>     },
>   }
> }
> ```
> 
> The package version is always available in the node context.

Oh interesting, yeah I was following the way it was done in the open-sauced repo. Will change it to this  🙂

**Comment by 0-vortex** [comment]
If you used the `gh` cli: 
```git restore package.json npm-shrinkwrap.json -s upstream/main```

If you forked normally:
```git restore package.json npm-shrinkwrap.json -s origin/main```

Open sauced lived to see Webpack, Snowpack and Vite - not everything is up to date there but it's usually better to ask when the docs are sketchy - took me a while to figure out the define plugin is used in Astro :D

**Comment by nulfrost** [comment]
For the __APP_VERSION__, it does work but eslint is complaining about it not being defined

**Comment by 0-vortex** [comment]
> > For the **APP_VERSION**, it does work but eslint is complaining about it not being defined
> 
> Oh yeah forgot about that one, needs to be added to eslint globals for it to work :D

add ```js
  globals: {
    __APP_VERSION__: 'readonly',
  },
``` to `.eslintrc.js` and that should fix it! 🍕

**Review by 0-vortex — APPROVED** [review]

**Pull request merged**